### DISCUSSION TITLE
Clustering of terminal voltage curves for a variation of parameters

### DISCUSSION THREAD

**Discussion by tGisbertz** [pybamm-team/PyBaMM#2566] [General]
Hello PyBaMM team,

I'm trying to do a parameter sensitivity analysis for my masterthesis. I started off with using the DFN model with the Chen2020 parameter set and I varied one parameter with each simulation to eventually compute the sensitivity with regard to that parameter. Looking at the resulting terminal voltage curves however I noticed that there are some strange clusters, even when I have an even distribution for the 10 parameter values I simulated (pictures and code example below).

Am I doing something wrong, or is the clustering somehow to be expected for some values?

![Discharge capacities for variations of Negative electrode active material volume fraction](https://user-images.githubusercontent.com/85396095/207867135-3f1cbe96-e26a-44b0-956a-015a4b3ca536.png)

![Discharge capacities for variations of Negative particle radius](https://user-images.githubusercontent.com/85396095/207868563-15e7e3c3-a770-4787-9819-8bcb47777de0.png)

Minimal example of my code:

```
#import packages
import pybamm
import numpy as np
import os


pybamm.set_logging_level("NOTICE")

#built battery model
model = pybamm.lithium_ion.DFN()

geometry = pybamm.battery_geometry(form_factor="cylindrical")
parameter_values = pybamm.ParameterValues("Chen2020")
parameter_values.process_geometry(geometry)
parameter_values.process_model(model)

#select number of points for discretisation
N = 40
N_r = 90
var_pts = {
    "x_n": N,  # negative electrode
    "x_s": N,  # separator 
    "x_p": N,  # positive electrode
    "r_n": N_r,  # negative particle
    "r_p": N_r,  # positive particle
} 

#define xperiment
experiment = pybamm.Experiment(period="0.5 s",
                                operating_conditions=
                                [("Discharge at 0.5C until 2.5V", "Rest for 10 minutes")]),

#parameters with value ranges
params_w_range = {      
      'Negative particle radius [m]': [1e-06, 2.2e-05],
      'Negative electrode active material volume fraction': [0.406, 0.75]
    }

new_output_variables = [
    "Terminal voltage [V]",
    "Time [s]",
    "Discharge capacity [A.h]",
    ]

#reset the parameter set after every simulation to only vary one parameter
parameter_values0 = parameter_values

solver = pybamm.CasadiSolver(mode="safe", atol=1e-6, rtol=1e-5, max_step_decrease_count=15)

for key in params_w_range:
    value_range = np.linspace(params_w_range[key][0], params_w_range[key][1], 10)    
    for value in value_range:
        # reset the parameter set
        parameter_values = parameter_values0
        parameter_values.update({key: value})
      
        name = ''.join(['0_5C', str(value).replace('.','_'), '.json'])
        sim = pybamm.Simulation(model, experiment=experiment, parameter_values=parameter_values, var_pts=var_pts, solver=solver)
        try:
            sim.solve([0, 3600], initial_soc=1)
        except:
            print(f'{name} failed to simulate')
            break
       
        sim.solution.save_data(filename=name, variables=new_output_variables, to_format='json')

```

**Comment by YoungAgile** [comment]
No one answered this question, and I am also working on related work. May I ask how you are doing, brother

**Comment by brosaplanella** [comment]
Sorry for the delay in the response, I completely missed this. Just a guess, but I think the issue might be that the parameters get modified in place, so when you change from modifying the first parameter to the second, the first is set to its final value.

**Reply by YoungAgile** [reply]
thanks for your reply
